lib/max_subarray.py

```diff
@@ -4,9 +4,29 @@ def max_sub_array(nums):
         Returns 0 if  nums is None or an empty list.
         Time Complexity: ?
         Space Complexity: ?
```

Comment:
✨ Your implementation works, but what about space and time complexity?

Reply:
Time Complexity: O(n)
 - The while loop has a linear time complexity since I am looping through the entire array once. The max function also has a linear time complexity. So 0(2n) simplifies to O(n)

Space Complexity: O(1)
 - There is no data structure being created or grown with this solution, so the space complexity remains constant